sdk/src/main/java/com/bugsnag/android/User.java

```diff
@@ -1,28 +1,23 @@
 package com.bugsnag.android;
 
+import android.content.SharedPreferences;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
 import java.io.IOException;
 
 /**
  * Information about the current user of your application.
  */
 class User implements JsonStream.Streamable {
-    private String id;
-    private String email;
-    private String name;
+    @Nullable private String id;
+    @Nullable private String email;
+    @Nullable private String name;
 
-    User() {
-    }
-
-    User(String id, String email, String name) {
-        this.id = id;
-        this.email = email;
-        this.name = name;
-    }
-
-    User(@NonNull User user) {
-        this(user.id, user.email, user.name);
+    User(Builder builder) {
```

Comment:
Can be private

Reply:
Making this private will generate a synthetic accessor call due to access from `User.Builder.build()` as there is no advantage to making private I'd prefer to avoid increasing the method count unnecessarily by one (once deprecated methods are removed, bugsnags method count be reduced even more which is always nice for a library).